packages/core/src/utilities/RLEVoxelMap.ts

```diff
@@ -0,0 +1,156 @@
+import isEqual from './isEqual';
+
+export type RLERun<T> = {
+  value: T;
+  i: number;
+  iEnd: number;
+  run: RLERun<T>;
+};
+
+/**
+ * RLE based implementation of a voxel map.
+ * This can be used as single or multi-plane, as the underlying indexes are
+ * mapped to rows and hte rows are indexed started at 0 and continuing
+ * incrementing for all rows in the multi-plane voxel.
+ */
+export default class RLEVoxelMap<T> {
```

Comment:
I really don't follow the code written in this class, as it needs various namings improvements and function documentations